index.html

```diff
@@ -311,6 +317,27 @@
   // This is where we'll need to create new event listeners for our filter and next/previous page buttons.
   // They'll look very similar to the above, just listening to different elements, and perhaps performing
   // different actions when clicked.
+
+  document.getElementById('filter-primary').addEventListener('click', function () {
+    if (options.filtered === 'primary') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'primary'
+    }
+
+    updateTable();
+  });
+
+  document.getElementById('filter-caucus').addEventListener('click', function () {
+    if (options.filtered === 'caucus') {
+      options.filtered = false;
+    } else {
+      options.filtered = 'caucus'
+    }
+
+    updateTable();
+  });
```

Comment:
This is great. Exactly what we want. For next week, try adding in a selected state (Bootstrap gives that to us if we add the class `active` to a button) using `d3.selectAll(...).classed('active', true/false)`.